src/hooks/use-create-software.ts

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import { useLambdasApi } from "src/hooks/use-api";
+import { SoftwareRegistry } from "src/generated/homeLambdasClient";
+
+const useCreateSoftware = (loggedUserId: string, setApplications: React.Dispatch<React.SetStateAction<SoftwareRegistry[]>>) => {
```

Comment:
set Applications can most likely be typed as () => void, please check. Also create an interface to define Props rather than do them inline here.